src/Razor/src/Microsoft.AspNetCore.Razor.LanguageServer/RazorRequestContextFactory.cs

```diff
@@ -27,7 +27,7 @@ public async Task<RazorRequestContext> CreateRequestContextAsync<TRequestParams>
         Uri? uri = null;
         if (textDocumentHandler is not null)
         {
-            if (textDocumentHandler is ITextDocumentIdentifierHandler<TRequestParams, TextDocumentIdentifier> tdiHandler)
+            if (textDocumentHandler is ITextDocumentIdentifierHandler<TRequestParams, VSTextDocumentIdentifier> tdiHandler)
```

Comment:
In line with above, I think this can be reverted, but then either the `TextDocumentIdentifier` returns needs to be passed to the `DocumentContext`, instead of the Uri, so it can be reused, or it can be cast to a `VSTextDocumentIdentifier` so the `ProjectContext` can be passed to the document context factory. Or something :)